### PR TITLE
net: wifi: shell: add reg domain support

### PR DESCRIPTION
**Pull request by Rex-Chen-NXP**
Supp api add reg domain support.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -1137,13 +1137,44 @@ int supplicant_reg_domain(const struct device *dev,
...
 		wpa_printf(MSG_ERROR, "Regulatory domain not supported");
 		return -ENOTSUP;
 	}
 
-	return wifi_mgmt_api->reg_domain(dev, reg_domain);
+	if (reg_domain->oper == WIFI_MGMT_GET) {
+		return wifi_mgmt_api->reg_domain(dev, reg_domain);
+	}
+
+	if (reg_domain->oper == WIFI_MGMT_SET) {
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:1152]: you can remove the the `if` as we return above for GET.

File: modules/hostap/src/supp_api.c [hunk 2]
```diff
@@ -1137,13 +1137,44 @@ int supplicant_reg_domain(const struct device *dev,
...
+
+		if (!wpa_cli_cmd_v("set country %s", reg_domain->country_code)) {
+			goto out;
+		}
+
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+		if (!hostapd_cli_cmd_v("set country_code %s", reg_domain->country_code)) {
+			goto out;
+		}
+#endif
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:1169]: should we do if/else i..e, only one command would be executed? or do we need both?

File: modules/hostap/src/supp_api.c (hunk 1 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c:1152]: Yes, of course,  but CI/CV check failed if change as comment.

File: modules/hostap/src/supp_api.c (hunk 2 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c:1169]: I think need both

Comment (krish2718) [reply modules/hostap/src/supp_api.c:1169]: for concurrent STA + AP case?

Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c:1169]: Yes, it is

File: modules/hostap/src/supp_api.c (hunk 1 above)
Comment (krish2718) [reply modules/hostap/src/supp_api.c:1152]: well, it should be the other way around :).

**Review by krish2718 — APPROVED** [review]

**Pull request merged**